flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/dataformat/BinaryString.java

```diff
@@ -654,531 +498,211 @@ private BinaryString trimSlow() {
-	 * keyvalue('k1:v1,k2:v2', ',', ':', 'k3') = NULL
-	 *
-	 * @param split1  separator between key-value tuple.
-	 * @param split2  separator between key and value.
-	 * @param keyName name of the key whose value you want return.
-	 *
-	 * @return target value.
-	 */
-	public BinaryString keyValue(byte split1, byte split2, BinaryString keyName) {
-		ensureMaterialized();
-		if (keyName == null || keyName.getSizeInBytes() == 0) {
-			return null;
-		}
-		if (inFirstSegment() && keyName.inFirstSegment()) {
-			// position in byte
-			int byteIdx = 0;
-			// position of last split1
-			int lastSplit1Idx = -1;
-			while (byteIdx < sizeInBytes) {
-				// If find next split1 in str, process current kv
-				if (segments[0].get(offset + byteIdx) == split1) {
-					int currentKeyIdx = lastSplit1Idx + 1;
-					// If key of current kv is keyName, return the value directly
-					BinaryString value = findValueOfKey(split2, keyName, currentKeyIdx, byteIdx);
-					if (value != null) {
-						return value;
-					}
-					lastSplit1Idx = byteIdx;
-				}
-				byteIdx++;
-			}
-			// process the string which is not ends with split1
-			int currentKeyIdx = lastSplit1Idx + 1;
-			return findValueOfKey(split2, keyName, currentKeyIdx, sizeInBytes);
-		} else {
-			return keyValueSlow(split1, split2, keyName);
-		}
-	}
+	// ------------------------------------------------------------------------------------------
+	// Internal methods on BinaryString
+	// ------------------------------------------------------------------------------------------
 
-	private BinaryString findValueOfKey(
-		byte split,
-		BinaryString keyName,
-		int start,
-		int end) {
-		int keyNameLen = keyName.sizeInBytes;
-		for (int idx = start; idx < end; idx++) {
-			if (segments[0].get(offset + idx) == split) {
-				if (idx == start + keyNameLen &&
-					segments[0].equalTo(keyName.segments[0], offset + start,
-						keyName.offset, keyNameLen)) {
-					int valueIdx = idx + 1;
-					int valueLen = end - valueIdx;
-					byte[] bytes = new byte[valueLen];
-					segments[0].get(offset + valueIdx, bytes, 0, valueLen);
-					return fromBytes(bytes, 0, valueLen);
-				} else {
-					return null;
-				}
-			}
-		}
-		return null;
+	byte getByteOneSegment(int i) {
+		return segments[0].get(offset + i);
 	}
 
-	private BinaryString keyValueSlow(
-		byte split1,
-		byte split2,
-		BinaryString keyName) {
-		// position in byte
-		int byteIdx = 0;
-		// position of last split1
-		int lastSplit1Idx = -1;
-		while (byteIdx < sizeInBytes) {
-			// If find next split1 in str, process current kv
-			if (getByte(byteIdx) == split1) {
-				int currentKeyIdx = lastSplit1Idx + 1;
-				BinaryString value = findValueOfKeySlow(split2, keyName, currentKeyIdx, byteIdx);
-				if (value != null) {
-					return value;
-				}
-				lastSplit1Idx = byteIdx;
-			}
-			byteIdx++;
-		}
-		int currentKeyIdx = lastSplit1Idx + 1;
-		return findValueOfKeySlow(split2, keyName, currentKeyIdx, sizeInBytes);
+	boolean inFirstSegment() {
```

Comment:
private?

Reply:
It is package access rights to let `BinaryStringUtil` to access it.